src/sources/aws_sqs/source.rs

```diff
@@ -123,17 +136,18 @@ impl SqsSource {
             match out.send_batch(events).await {
                 Ok(()) => {
                     if self.delete_message {
-                        if let Some(receiver) = batch_receiver {
-                            let client = self.client.clone();
-                            let queue_url = self.queue_url.clone();
-                            tokio::spawn(async move {
-                                let batch_status = receiver.await;
-                                if batch_status == BatchStatus::Delivered {
-                                    delete_messages(&client, &receipts_to_ack, &queue_url).await;
-                                }
-                            });
-                        } else {
-                            delete_messages(&self.client, &receipts_to_ack, &self.queue_url).await;
+                        match batch_receiver {
+                            Some(receiver) => finalizer
+                                .unwrap_or_else(|| unreachable!())
```

Comment:
Oops, yeah, that should have a message. I'll fix that.

As for the difference in the generated code, [the answer is, not much](https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=306a187e3c4c432f3e695ed4d0829ead). Effectively it results in identical code paths on the normal path (at least in release mode), just a difference in what code is generated for the failure path. The `unwrap`+`unreachable` combination puts the panic inline (with a different message) instead of panicking from within the `expect`. Probably not enough difference to be worth caring.